torch/nativert/executor/Placement.h

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include <c10/core/Device.h>
+#include <c10/util/Logging.h>
+
+#include <optional>
+#include <unordered_map>
+
+namespace torch::nativert {
+
+/**
+ * @brief Normalizes a c10::Device by ensuring a canonical form.
+ *
+ * This function returns a normalized version of the input device:
+ * - For CPU devices, the returned device will have no index (i.e., the default CPU device).
+ * - For CUDA devices, if no index is specified, index 0 is assumed.
+ * - For other device types, the function will raise an error.
+ *
+ * @param device The input c10::Device to normalize.
+ * @return A normalized c10::Device with standardized indexing.
+ *
+ * @throws c10::Error If the device type is not CPU or CUDA.
+ */
+
+TORCH_API c10::Device normalizeDevice(const c10::Device& device);
+
+/**
+ * Returns true if the two devices are the same and has the same device index (if cuda).
+ */
+TORCH_API bool isSameDevice(
+    const c10::Device& device1,
+    const c10::Device& device2);
+
+
+/**
+ * @brief A utility class for managing device placement mappings.
+ *
+ * The Placement class provides a way to map source devices to target devices.
+ * It supports both explicit per-device mappings and a default device fallback.
+ * This is used for device parameterization in Torch Native Runtime, and to support
+ * execution with various backends, or with various device placement.
+ */
+struct TORCH_API Placement {
```

Comment:
Do you need this to be public?

Reply:
@albanD Yeah I think this is part of the nativert API. I agree all the other things might not need to be TORCH_API, but Placement is something user provide to us to define where to execute the model, according to the RFC https://github.com/pytorch/rfcs/blob/1175e55852c9a95b3131607a04d184055e37ebde/RFC-0043-torch-native-runtime.md#proposed-implementation

Reply:
Sounds good!

Reply:
Users will might need to construct their Placement and use it as an input to the runtime, so I think we might need to make it public?

Reply:
@albanD I think this Placement needs to be public because people might need to construct their own Placement object and pass it into the runtime.